gmso/lib/potential_templates.py

```diff
@@ -1,158 +1,130 @@
+import glob
+import os
+import json
+
 from gmso.core.potential import Potential
+from gmso.utils.singleton import Singleton
+from gmso.exceptions import GMSOError
 
+POTENTIAL_JSONS = glob.glob(os.path.join(os.path.dirname(__file__), 'jsons', '*.json'))
+JSON_DIR = os.path.join(os.path.dirname(__file__), 'jsons')
+USER_TEMPLATE_PREFIX = 'potential_templates'
 
-class PotentialTemplate(Potential):
-    def __init__(self,
-                 name='PotentialTemplate',
-                 expression='4*epsilon*((sigma/r)**12 - (sigma/r)**6)',
-                 independent_variables={'r'},
-                 template=True):
 
-        super(PotentialTemplate, self).__init__(
-            name=name,
-            expression=expression,
-            independent_variables=independent_variables,
-            template=template,
-        )
+def _create_hidden_dir(new_dir=None):
+    gmso_dir = os.path.join(os.path.expanduser('~'), '.gmso')
```

Comment:
I feel like this will work for 90% of users, but I also wonder if there is already a best-practices way of doing this? It seems like something that is nearly ubiquitous, for which there may be some community standard.